dev/kyuubi-extension-spark-3-1/src/main/scala/org/apache/kyuubi/sql/DropIgnoreNonexistent.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kyuubi.sql
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.plans.logical.{DropTable, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.execution.command.{AlterTableDropPartitionCommand, DropDatabaseCommand, DropFunctionCommand, DropTableCommand}
+
+import org.apache.kyuubi.sql.KyuubiSQLConf._
+
+case class DropIgnoreNonexistent(session: SparkSession) extends Rule[LogicalPlan] {
+
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    if (conf.getConf(DROP_IGNORE_NONEXISTENT)) {
+      plan match {
+        case i @ AlterTableDropPartitionCommand(tableName, specs, ifExists, purge, retainData)
+          if !ifExists =>
+          i.copy(tableName, specs, true, purge, retainData)
+        case i @ DropTable(child, ifExists, purge) if !ifExists =>
```

Comment:
Why we need catch `DropTable` ?

Reply:
> Why we need catch `DropTable` ?

When the table does not exist, it does not seem to be resolved as a DropTableCommand.

Reply:
ah I see. It seems not compatiable between Spark 3.1 and 3.2.
- For 3.1, `DropTable` ifExists -> `NoopDropTable `
- For 3.2, `DropTable` ifExists -> `ResolveCommandsWithIfExists`

So  it looks more better to me if we can add this rule using `injectPostHocResolutionRule` like `NoopDropTable` or `ResolveCommandsWithIfExists`. What do you think ?

Reply:
Thanks @ulysses-you , it sounds good to me, I will try it.